packages/engine/src/store/shared/addNode.ts

```diff
@@ -4,62 +4,99 @@ import { createUniqueId } from '@utils/createUniqueId'
 const vector3Keys = ['x', 'y', 'z']
 const rgbKeys = ['r', 'g', 'b']
 
-export const addNode = (
+const _addNodeToState = (
   state: EngineState,
   paramId: string,
   sketchId: string,
-  { key, valueType = NodeTypes.Number, defaultValue }: SketchConfigParam,
+  { key, defaultValue, valueType, ...config }: SketchConfigParam,
 ) => {
   if (isNodeTypeWithChildren(valueType)) {
-    if (!Array.isArray(defaultValue)) {
-      throw new Error(`Expected defaultValue to be an array for ${valueType} type`)
+    throw new Error(`addNodeAndOptionNodes shouldn't be used with node of valueType: ${valueType}`)
+  }
+
+  state.nodes[paramId] = {
+    id: paramId,
+    key,
+    type: 'param' as const,
+    sketchId,
+    valueType,
+    ...config,
+  }
+
+  if (
+    (typeof defaultValue === 'number' && valueType === NodeTypes.Number) ||
+    (typeof defaultValue === 'boolean' && valueType === NodeTypes.Boolean) ||
+    (typeof defaultValue === 'string' && valueType === NodeTypes.Enum)
+  ) {
+    state.nodeValues[paramId] = defaultValue
+  } else {
+    throw new Error(`valueType of param ${key} does not match defaultValue for sketch ${sketchId}`)
+  }
+
+  return state.nodes[paramId]
+}
+
+const _addOptionNodeToState = (
+  state: EngineState,
+  paramId: string,
+  sketchId: string,
+  sketchConfigParam: SketchConfigParam,
+) => {
+  /** TODO: This can probably be tidier, using some sort of config object to generate the option nodes
+   * The same config object could also be used in the component to loop through
+   */
+  if (sketchConfigParam.valueType === NodeTypes.Number) {
+    _addNodeToState(state, `${paramId}-sliderMin`, sketchId, {
+      key: 'sliderMin',
+      valueType: NodeTypes.Number,
+      defaultValue: sketchConfigParam.sliderMin ?? 0,
```

Comment:
so configs need to come with sliderMin/Max instead of defaultMin/Max now? Not an issue, just want to make sure I've got it right, might be good to throw that info in an md file too

Reply:
Also is defaultValue the raw value, or normalized between the sliderMin/Max